oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/property/PropertyIndexCleaner.java

```diff
@@ -97,6 +100,35 @@ public void run() {
         }
     }
 
+    /**
+     * Perform some cleanup.
+     *
+     * @param paths the list of paths (comma separated)
+     * @param batchSize the bach size
+     * @param sleepPerBatch the number of milliseconds to sleep per batch
+     * @param maxRemoveCount the maximum number of nodes to remove per path
+     * @return the number of nodes removed
+     */
+    public int performCleanup(String paths, int batchSize, int sleepPerBatch, int maxRemoveCount) throws CommitFailedException {
+        ArrayList<String> list = new ArrayList<>(Arrays.asList(paths.split(",")));
+        int numOfNodesDeleted = 0;
+        for(String s : list) {
+            log.info("Cleanup of {}", s);
+            if (!NodeStateUtils.isHidden(PathUtils.getName(s))) {
+                log.warn("Not a hidden node");
+                continue;
+            }
+            RecursiveDelete rd = new RecursiveDelete(nodeStore, createCommitHook(),
+                    PropertyIndexCleaner::createCommitInfo);
+            rd.setBatchSize(batchSize);
+            rd.setSleepPerBatch(sleepPerBatch);
+            rd.setMaxRemoveCount(maxRemoveCount);
+            rd.run(Arrays.asList(s));
```

Comment:
this could be simplified with:
```suggestion
            rd.run(singletonList(s));
```